### PR TITLE
Shadowshrooms + Glowshroom Nerf + Station Lighting Nerf

### PR DESCRIPTION
**Pull request by uraniummeltdown**
https://github.com/tgstation/tgstation/pull/24743
https://github.com/tgstation/tgstation/pull/24898

Here are 100 potency glowshrooms and 100 potency shadowshrooms in a well lit area
shadow

Moved light color from the glowshroom effect into the glow trait, added new genes for red light and "dark light". Shadowshrooms are a new glowshroom mutation which emit shadows. Nerfed the glowshroom glow a bit.

I turned down the brightness_power of station tubes and bulbs from 3 and 2 down to 1. It's not  noticeable except in maint and 1 bulb rooms.

:cl:
tweak: Station lighting has less lighting power, this is not noticeable apart from maint and 1 bulb rooms being a bit darker.
tweak: Tweaked the potency scaling for glowshroom/glowberry light; high-potency plants no longer light up a huge area, but are slightly brighter.
add: Added Shadowshrooms as a glowshroom mutation. They do exactly what you'd expect.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Oh, I was going to do this. You beat me to it.

Thank you.

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 1]
```diff
@@ -292,3 +302,26 @@
...
+	desc = "This mycelium will grow into something shadowy."
+	icon_state = "mycelium-shadowshroom"
+	species = "shadowshroom"
+	icon_grow = "shadowshroom-grow"
+	icon_dead = "shadowshroom-dead"
+	plantname = "Shadowshrooms"
+	product = /obj/item/weapon/reagent_containers/food/snacks/grown/mushroom/glowshroom/shadowshroom
+	genes = list(/datum/plant_gene/trait/glow/shadow, /datum/plant_gene/trait/plant_type/fungal_metabolism)
+	mutatelist = list()
+	reagents_add = list("radium" = 0.2, "nutriment" = 0.04)
```
Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/mushrooms.dm]: plant matter instead of nutriment.

Comment (uraniummeltdown) [inline code/modules/hydroponics/grown/mushrooms.dm]: The other mushrooms in this file also have nutriment, should I change those too?

Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/mushrooms.dm]: Oh. Yeah, leave these be, nutriment should still be a *thing* in some plants. If the others don't have plant matter, these shouldn't either.

**Comment by KingPhilipIII** [comment]
How to win as a sling. Thrall the botanist.

**Comment by taukausanake** [comment]
No way to get different colored glows yet? I would enjoy that at some point.

Looks more purple than shadow. Explain please?

**Comment by jughu** [comment]
Whentheres alot of shadowshrooms next to eachother the purple stacks creating a dark tole

**Comment by uraniummeltdown** [comment]
Do not merge yet I might need to tweak it

**Comment by Fox-McCloud** [comment]
Yeah, there's been 2-4 tweaks to glow and shadowshrooms (strength, color, and scaling), that would probably be important to add to this.

**Comment by uraniummeltdown** [comment]
So I tested tg's shadowshroom darkness nerf and the shadowshrooms are hardly noticeable, I looked at our lighting and our /obj/machinery/light has brightness_power 3 while tg has it set to 1 (probably why they nerfed the shadows). Personally I think our lighting is a bit too bright, should I change brightness_power to 2 or 1 or should I not go ahead with tg's shadowshroom nerf and leave it like the OP?

**Comment by uraniummeltdown** [comment]
@Fox-McCloud Well I went ahead and nerfed lighting too, light_power (intensity) of lighting reduced to 1 from 3 for tube and 2 for bulb and also applied the shadowshroom glow nerf, this is what it looks like. Should I keep it like this or should I unnerf both lighting and shadowshrooms so that it's more like OP image? 
shadowshroom

File: code/game/objects/effects/glowshroom.dm [hunk 2]
```diff
@@ -10,30 +10,54 @@
...
+		myseed = newseed.Copy()
+		myseed.forceMove(src)
+	else
+		myseed = new myseed(src)
+	if(mutate_stats) //baby mushrooms have different stats :3
+		myseed.adjust_potency(rand(-3,6))
+		myseed.adjust_yield(rand(-1,2))
+		myseed.adjust_production(rand(-3,6))
+		myseed.adjust_endurance(rand(-3,6))
+	delay = delay - myseed.production * 100 //So the delay goes DOWN with better stats instead of up. :I
```
Comment (Fox-McCloud) [inline code/game/objects/effects/glowshroom.dm]: `endurance = myseed.endurance`

**Comment by Fox-McCloud** [comment]
Once you adjust the endurance/integrity of the glowshroom object, this will be good to go.

**Comment by Fox-McCloud** [comment]
@uraniummeltdown

**Comment by uraniummeltdown** [comment]
@Fox-McCloud Done the endurance thing. Also updated PR title and changelog.

**Pull request merged**